checks/ulimits.go

```diff
@@ -0,0 +1,121 @@
+package checks
+
+import (
+	"fmt"
+	"syscall"
+)
+
+// UlimitChecker checks if current process resource limits are above a given minimum
+//
+// Type:
+//   - ulimit
+//
+// Supported Platforms:
+//   - MacOS
+//   - Linux
+//
+// Arguments:
+//   - item (required): A string value representing the type of limit to check
+//   - limit (required): Numerical value representing the minimum value to be tested
+//   - type (optional): "hard" or "soft" with a default of "hard"
```

Comment:
nit: our optionals on all other checks don't have (optional) they just omit (required), so this is mildly inconsistent.

Reply:
We can discuss if this is right or wrong if you want to make a case

Reply:
sgtm, I just removed it